### PR TITLE
rgw/multisite: fix shard count while fetching bucket sync status after reshard

### PR DESCRIPTION
**Pull request by smanjara**
Signed-off-by: Shilpa Jagannath <smanjara@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -2319,6 +2315,12 @@ static int bucket_source_sync_status(const DoutPrefixProvider *dpp, rgw::sal::Ra
     return r;
   }
 
+  status.resize(remote_markers.get().size());
```
Comment (cbodley) [inline src/rgw/rgw_admin.cc:2318]: we called `rgw_read_bucket_full_sync_status()` above, so we can use `full_status.shards_done_with_gen.size()` as the local num_shards

we should probably also compare that against `remote_markers.get().size()`, and return EINVAL if the local and remote num_shards don't match

Comment (smanjara) [reply src/rgw/rgw_admin.cc:2318]: both come from remote. we can use any one of them, right?

Comment (smanjara) [reply src/rgw/rgw_admin.cc:2318]: if we call  bucket_sync_status while `shards_done_with_gen` is not fully populated, we will still end up crashing.

**Comment by cbodley** [comment]
this merged with https://github.com/ceph/ceph/pull/42872

**Pull request closed without merging**